mem/table.go

```diff
@@ -2,86 +2,279 @@ package mem
 
 import (
 	"bytes"
-	"encoding/binary"
+	"encoding/gob"
 	"fmt"
 	"io"
+	"strconv"
 
-	errors "gopkg.in/src-d/go-errors.v1"
+	errors "gopkg.in/src-d/go-errors.v0"
 	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
 )
 
 // Table represents an in-memory database table.
 type Table struct {
-	name   string
-	schema sql.Schema
-	data   []sql.Row
+	name       string
+	schema     sql.Schema
+	partitions map[string][]sql.Row
+	keys       [][]byte
+
+	insert int
+
+	Filters []sql.Expression
+	Columns []int
+	Lookup  sql.IndexLookup
 }
 
+var _ sql.Table = (*Table)(nil)
+var _ sql.Inserter = (*Table)(nil)
+var _ sql.FilteredTable = (*Table)(nil)
+var _ sql.ProjectedTable = (*Table)(nil)
+var _ sql.IndexableTable = (*Table)(nil)
+
 // NewTable creates a new Table with the given name and schema.
-func NewTable(name string, schema sql.Schema) *Table {
+func NewTable(name string, schema sql.Schema, numPartitions int) *Table {
```

Comment:
since most of the calls are with 0 numPartitions, how about having `NewTable` and `NewTablePartitioned` for when you want partitions?